packages/kbn-config-schema/src/types/array_type.test.ts

```diff
@@ -188,3 +188,29 @@ describe('#maxSize', () => {
     ).toThrowErrorMatchingInlineSnapshot(`"array size is [2], but cannot be greater than [1]"`);
   });
 });
+
+describe('#extendsDeep', () => {
+  const type = schema.arrayOf(
+    schema.object({
+      foo: schema.string(),
+    })
+  );
+
+  test('objects with unknown attributes are kept when extending with unknowns=allow', () => {
+    const result = type.extendsDeep({ unknowns: 'allow' }).validate([{ foo: 'test', bar: 'test' }]);
```

Comment:
Good point! I updated the description and tests (https://github.com/elastic/kibana/pull/156214/commits/fc31e3285c1fcefa3d94b0892204b54b5831c0c8) so the bad way to do it can remain our little secret.